src/app/fetch-activity.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { delay, from, Observable, of } from 'rxjs';
+
+import allContextDecksCollection from '../assets/activities/contextarray.json';
+import allCardDecksCollection from '../assets/activities/cardsarray.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class FetchActivityService {
+  importedData!: [];
+  activityData!: any;
+
+  constructor() { }
+
+  fetchActivity(segment: string, Id: number): Observable<any> {
```

Comment:
More functions instead of 1 @leventemo

Reply:
Created separate fns for all activities, although there's still some code repetition going on in the RXJS bit.